src/controller/signup.rs

```diff
@@ -1,6 +1,40 @@
+use sha3::{Digest, Sha3_256};
+
+use crate::app_context::AppContext;
+use crate::usecase;
 use crate::{views, views::TemplateToResponse};
-use actix_web::Responder;
+use actix_web::{web, HttpResponse, Responder};
+use serde::Deserialize;
+use validator::Validate;
 
 pub async fn signup() -> impl Responder {
     views::signup::SignUpTemplate {}.to_response()
 }
+
+pub async fn signup_post(req: web::Form<Req>, context: web::Data<AppContext>) -> impl Responder {
+    match req.validate() {
+        Ok(_) => (),
+        Err(_) => return HttpResponse::BadRequest().body("validate error"),
+    }
```

Comment:
[imo]

もうちょっとスッキリ書けそうですね

```suggestion
    if req.validate().is_err() {
        return HttpResponse::BadRequest().body("validate error"),
    }
```

ちなみに `Err` の中身を取り出したいときは `if let` 構文も使えます

```rust
if let Err(e) = req.validate() {

}
```